File: backend/lib/L1_domain/entities/goals/person.py
#  Copyright (c) 2022. Alexandr Moroz

from dataclasses import dataclass

from ..auth.workspace_bounded import WorkspaceBounded
from ..base_entity import Emailable, Persistable


@dataclass
class Person(Persistable, Emailable, WorkspaceBounded):
    firstname: str | None = None
    lastname: str | None = None
    email: str | None = None

File: lib/L2_data/mappers/person.dart
// Copyright (c) 2022. Alexandr Moroz

import 'package:openapi/openapi.dart' as api;

import '../../L1_domain/entities/person.dart';

extension PersonMapper on api.PersonGet {
  Person person(int wsId) => Person(
        id: id,
        email: email,
        firstname: firstname,
        lastname: lastname,
        workspaceId: wsId,
      );
}
